code/modules/reagents/chemistry/machinery/chem_master.dm

```diff
@@ -620,20 +496,148 @@
 	component_parts += new /obj/item/reagent_containers/glass/beaker(null)
 	RefreshParts()
 
+/datum/chemical_production_mode
+	var/production_name = ""
+	/// FontAwesome icon name
+	var/production_icon = ""
+	var/obj/item/reagent_containers/item_type
+	var/list/sprites
+	var/max_items_amount = 0
+	var/max_units_per_item = 0
+	var/name_suffix = ""
+
+	var/set_sprite = 1
+	var/set_name = ""
+	var/set_items_amount = 1
+	var/placeholder_name = ""
+
+/datum/chemical_production_mode/proc/PlaceholderName(datum/reagents/reagents)
```

Comment:
Please use snake_case for proc names, and I think adding `get` before it here is good if you're trying to get the name
```suggestion
/datum/chemical_production_mode/proc/get_placeholder_name(datum/reagents/reagents)
```